src/HPolyhedron.jl

```diff
@@ -484,11 +484,13 @@ function remove_redundant_constraints!(P::PT;
         br = b[non_redundant_indices]
         br[i] = b[j] + one(N)
         lp = linprog(-α, Ar, '<', br, -Inf, Inf, backend)
-        if lp.status != :Optimal
-            error("LP is not optimal")
+        if lp.status == :Infeasible
+            throw(DomainError(P, "the polyhedron is empty"))
+        else
+            error("LP is not optimal; the status of the LP is $(lp.status)")
```

Comment:
what type of error should be thrown here?

Reply:
Why does it matter?

Reply:
thinking of a beter practice, because it helps to debug for example:

``` try
              remove_redundant_constraints!(P)
       catch y
           if isa(y, DomainError)
                  return EmptySet()
           elseif isa(y, BoundsError)
                   . . . .
           else  
                   ...
            end
       end
```

Reply:
Then you have to study all cases when this can happen. I would say that this is not worth the effort.